src/pytorch_lightning/tuner/tuning.py

```diff
@@ -62,14 +69,38 @@ def _tune(
         if self.trainer.auto_scale_batch_size:
             if isinstance(self.trainer.auto_scale_batch_size, str):
                 scale_batch_size_kwargs.setdefault("mode", self.trainer.auto_scale_batch_size)
-            result["scale_batch_size"] = scale_batch_size(self.trainer, model, **scale_batch_size_kwargs)
+
+            result["scale_batch_size"] = self.scale_batch_size(
+                model, train_dataloaders, val_dataloaders, dataloaders, datamodule, method, **scale_batch_size_kwargs
+            )
 
         # Run learning rate finder:
         if self.trainer.auto_lr_find:
+            self.trainer.state.fn = TrainerFn.TUNING
```

Comment:
Should we think of removing `TrainerFn.TUNING`? Since it can call all of the other trainer functions internally

Reply:
yes, in a follow-up. Since lr_finder wasn't merged, I didn't remove it here.

Reply:
added a TODO.